xapian-bindings/java/SmokeTest.java

```diff
@@ -142,6 +154,15 @@ public static void main(String[] args) throws Exception {
 		System.err.println("Unexpected docid");
 		    System.exit(1);
 	    }
+	    // New iterator, should start from the beginning 
```

Comment:
There's a trailing space on this line. Two ways of spotting these: `git show` will generally highlight trailing spaces (and tabs mixed with spaces); also running `git diff origin/master | ./xapian-maintainer-tools/xapian-check-patch` from the top of the checkout will check for these common whitespace problems. (You can also do `git show | ./xapian-maintainer-tools/xapian-check-patch` if you've already created a single commit.)

Right now, `xapian-check-patch` is showing about a dozen errors. If you can get them fixed, and correct the test (did you actually run the smoketest? make sure you do and that it passes) then this should be ready for merge.

If you can make changes as fixup commits, and once we're ready we can squash them down.